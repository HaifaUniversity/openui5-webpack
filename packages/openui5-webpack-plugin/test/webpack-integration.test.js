'use strict';

const fs = require('fs');
const path = require('path');
const webpack = require('webpack');
const OpenUI5Plugin = require('../src/OpenUI5Plugin');

const cases = fs.readdirSync(path.join(__dirname, 'cases'));
describe('Integration TestCases', () => {
  cases.forEach((testCase) => {
    it(testCase, (done) => {
      let options = {
        entry: {
          build: './index.js',
        },
        module: {
          rules: [
            {
              test: /\.xml$/,
              use: 'xml-loader',
            },
          ],
        },
        plugins: [
          new OpenUI5Plugin(),
        ],
        mode: 'production'
      };

      const testDirectory = path.join(__dirname, 'cases', testCase);
      const outputDirectory = path.join(__dirname, 'js', testCase);
      const configFile = path.join(testDirectory, 'webpack.config.js');

      if (fs.existsSync(configFile)) {
        options = require(configFile);
      }

      if (!options.context) options.context = testDirectory;
      if (!options.output) options.output = { filename: '[name].js' };
      if (!options.output.path) options.output.path = outputDirectory;

      webpack(options, (err, stats) => {
        if (err) {
          done(err);
          return;
        }
        if (stats.hasErrors()) {
          done(new Error(stats.toString()));
          return;
        }

        const actualPath = path.join(outputDirectory, 'build.js');
        expect(readFileOrEmpty(actualPath)).toMatchSnapshot();
        done();
      });
    });
  });
});

function readFileOrEmpty(path) {
  try {
    return fs.readFileSync(path, 'utf-8');
  } catch (e) {
    return '';
  }
}
