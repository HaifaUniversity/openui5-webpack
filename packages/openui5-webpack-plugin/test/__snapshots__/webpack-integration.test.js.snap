// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration TestCases define-dep-fcn 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_ARRAY__, __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_ARRAY__ = [__webpack_require__(1)], __WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.apply(exports, __WEBPACK_UI5_DEFINE_ARRAY__),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-dep-fcn-exp 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_ARRAY__, __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_ARRAY__ = [__webpack_require__(1)], __WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.apply(exports, __WEBPACK_UI5_DEFINE_ARRAY__),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-fcn 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.call(exports, __webpack_require__, exports, module),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-fcn-exp 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.call(exports, __webpack_require__, exports, module),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-name-dep-fcn 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_ARRAY__, __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_ARRAY__ = [__webpack_require__(1)], __WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.apply(exports, __WEBPACK_UI5_DEFINE_ARRAY__),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-name-dep-fcn-exp 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_ARRAY__, __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_ARRAY__ = [__webpack_require__(1)], __WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.apply(exports, __WEBPACK_UI5_DEFINE_ARRAY__),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases define-name-fcn 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var __WEBPACK_UI5_DEFINE_RESULT__;!(__WEBPACK_UI5_DEFINE_RESULT__ = function() {
  return 'test';
}.call(exports, __webpack_require__, exports, module),
        __WEBPACK_UI5_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_UI5_DEFINE_RESULT__));


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases globals 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

(__webpack_require__(1)).test();


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports.test = function() {};


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases jquery-require 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(0);
__webpack_require__(0);


/***/ })
],[[1,1]]]);"
`;

exports[`Integration TestCases jquery-unknown-context 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

const a = 'a';
console.warn(\\"UI5 tried to dynamically require a module. If it doesn't exist in the bundle a error might follow.\\");


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases require-array 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[1],{

/***/ 2:
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__.e(/* openui5 require */ 0).then(function() { var __WEBPACK_UI5_REQUIRE_ARRAY__ = [__webpack_require__(0), __webpack_require__(1)]; (function(a, b) {
  const c = b;
}).apply(null, __WEBPACK_UI5_REQUIRE_ARRAY__);}).catch(__webpack_require__.oe);

__webpack_require__.e(/* openui5 require */ 0).then(function() { var __WEBPACK_UI5_REQUIRE_ARRAY__ = [__webpack_require__(0), __webpack_require__(1)];}).catch(__webpack_require__.oe);


/***/ })

},[[2,2]]]);"
`;

exports[`Integration TestCases require-context 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'a';
__webpack_require__(2)(\\"./\\" + a);


/***/ }),
/* 2 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./a\\": 0,
	\\"./a.js\\": 0
};


function webpackContext(req) {
	var id = webpackContextResolve(req);
	return __webpack_require__(id);
}
function webpackContextResolve(req) {
	var id = map[req];
	if(!(id + 1)) { // check for number or string
		var e = new Error(\\"Cannot find module '\\" + req + \\"'\\");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	}
	return id;
}
webpackContext.keys = function webpackContextKeys() {
	return Object.keys(map);
};
webpackContext.resolve = webpackContextResolve;
module.exports = webpackContext;
webpackContext.id = 2;

/***/ })
],[[1,1]]]);"
`;

exports[`Integration TestCases require-item 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases require-sync 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases resource 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'a.xml';
__webpack_require__(1)((a).replace(/^(?:\\\\.\\\\/)?/, \\".\\"), true);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./context/a.xml\\": 2
};
function openui5Resource(req, failOnError) {
  var id = openui5ResourceResolve(req, failOnError);
  if (id === null) {
    return null;
  }
  if(!__webpack_require__.m[id])
    throw new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
  return __webpack_require__(id);
};
function openui5ResourceResolve(req, failOnError) {
  var id = map[req];
  if(!(id + 1)) { // check for number or string
    if (failOnError) {
      throw new Error(\\"Cannot find module '\\" + req + \\"'.\\");
    }
    return null;
  }
  return id;
};
openui5Resource.keys = function openui5ResourceKeys() {
  return Object.keys(map);
};
openui5Resource.resolve = openui5ResourceResolve;
openui5Resource.id = 1;
module.exports = openui5Resource;

/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases resource-fail-on-error 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'sap/ui/test/a.xml';
__webpack_require__(1)((a).replace(/^(?:\\\\.\\\\/)?sap\\\\/ui\\\\/test/, \\".\\"), false);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./a.xml\\": 2
};
function openui5Resource(req, failOnError) {
  var id = openui5ResourceResolve(req, failOnError);
  if (id === null) {
    return null;
  }
  if(!__webpack_require__.m[id])
    throw new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
  return __webpack_require__(id);
};
function openui5ResourceResolve(req, failOnError) {
  var id = map[req];
  if(!(id + 1)) { // check for number or string
    if (failOnError) {
      throw new Error(\\"Cannot find module '\\" + req + \\"'.\\");
    }
    return null;
  }
  return id;
};
openui5Resource.keys = function openui5ResourceKeys() {
  return Object.keys(map);
};
openui5Resource.resolve = openui5ResourceResolve;
openui5Resource.id = 1;
module.exports = openui5Resource;

/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases resource-library 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'sap/ui/test/a.xml';
__webpack_require__(1)((a).replace(/^(?:\\\\.\\\\/)?sap\\\\/ui\\\\/test/, \\".\\"), true);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./a.xml\\": 2,
	\\"./sap/m/messagebundle.properties\\": 6,
	\\"./sap/ui/core/messagebundle.properties\\": 4,
	\\"./sap/ui/core/messagebundle_en.properties\\": 5,
	\\"sap/ui/core/cldr/en.json\\": 3
};
function openui5Resource(req, failOnError) {
  var id = openui5ResourceResolve(req, failOnError);
  if (id === null) {
    return null;
  }
  if(!__webpack_require__.m[id])
    throw new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
  return __webpack_require__(id);
};
function openui5ResourceResolve(req, failOnError) {
  var id = map[req];
  if(!(id + 1)) { // check for number or string
    if (failOnError) {
      throw new Error(\\"Cannot find module '\\" + req + \\"'.\\");
    }
    return null;
  }
  return id;
};
openui5Resource.keys = function openui5ResourceKeys() {
  return Object.keys(map);
};
openui5Resource.resolve = openui5ResourceResolve;
openui5Resource.id = 1;
module.exports = openui5Resource;

/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases resource-object 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'a.xml';
__webpack_require__(1)((a).replace(/^(?:\\\\.\\\\/)?/, \\".\\"), true);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./context/a.xml\\": 2
};
function openui5Resource(req, failOnError) {
  var id = openui5ResourceResolve(req, failOnError);
  if (id === null) {
    return null;
  }
  if(!__webpack_require__.m[id])
    throw new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
  return __webpack_require__(id);
};
function openui5ResourceResolve(req, failOnError) {
  var id = map[req];
  if(!(id + 1)) { // check for number or string
    if (failOnError) {
      throw new Error(\\"Cannot find module '\\" + req + \\"'.\\");
    }
    return null;
  }
  return id;
};
openui5Resource.keys = function openui5ResourceKeys() {
  return Object.keys(map);
};
openui5Resource.resolve = openui5ResourceResolve;
openui5Resource.id = 1;
module.exports = openui5Resource;

/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases resource-path-replace 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const a = 'sap/ui/test/a.xml';
__webpack_require__(1)((a).replace(/^(?:\\\\.\\\\/)?sap\\\\/ui\\\\/test/, \\".\\"), true);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./a.xml\\": 2
};
function openui5Resource(req, failOnError) {
  var id = openui5ResourceResolve(req, failOnError);
  if (id === null) {
    return null;
  }
  if(!__webpack_require__.m[id])
    throw new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
  return __webpack_require__(id);
};
function openui5ResourceResolve(req, failOnError) {
  var id = map[req];
  if(!(id + 1)) { // check for number or string
    if (failOnError) {
      throw new Error(\\"Cannot find module '\\" + req + \\"'.\\");
    }
    return null;
  }
  return id;
};
openui5Resource.keys = function openui5ResourceKeys() {
  return Object.keys(map);
};
openui5Resource.resolve = openui5ResourceResolve;
openui5Resource.id = 1;
module.exports = openui5Resource;

/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases view 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

new (__webpack_require__(1))(undefined);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = 'test';


/***/ })
],[[0,1]]]);"
`;

exports[`Integration TestCases view-context 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

var a = \\"sap/ui/core/mvc/DummyView\\";
new (__webpack_require__(1)(a.replace(\\"sap/ui/core/mvc\\", \\".\\") + \\".js\\"))(oView);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./DummyView.js\\": 2
};


function webpackContext(req) {
	var id = webpackContextResolve(req);
	if(!__webpack_require__.m[id]) {
		var e = new Error(\\"Module '\\" + req + \\"' ('\\" + id + \\"') is not available (weak dependency)\\");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	}
	return __webpack_require__(id);
}
function webpackContextResolve(req) {
	var id = map[req];
	if(!(id + 1)) { // check for number or string
		var e = new Error(\\"Cannot find module '\\" + req + \\"'\\");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	}
	return id;
}
webpackContext.keys = function webpackContextKeys() {
	return Object.keys(map);
};
webpackContext.resolve = webpackContextResolve;
webpackContext.id = 1;
module.exports = webpackContext;

/***/ })
],[[0,1]]]);"
`;
